src/libcamera/pipeline/rpi/pisp/pisp.cpp

```diff
@@ -2058,6 +2053,16 @@ int PiSPCameraData::configureEntities(V4L2SubdeviceFormat sensorFormat,
 	if (ret)
 		return ret;
 
+	if (sensorMetadata_) {
+		ret = csi2Subdev_->setFormat(csiMetaSinkPad, &embeddedFormat);
```

Comment:
Setting this seems like window-dressing.  Does it give us any actual benefit?

Reply:
Both the sink side and the source side of a v4l2 subdev has to be configured, otherwise it won't work correctly (or it works only by luck, if the default format happens to be a correct one). It's no different for metadata than for video.